### DISCUSSION TITLE
Hook that triggers before response model validation

### DISCUSSION THREAD

**Discussion by wackazong** [fastapi/fastapi#6909] [Questions] [answered]
Ist it possible to have a method running after each request but _before_ the response model validation? I tried middleware and APIRoute as described in the docs, they are both after response model validation.

**Comment by alexschoeberl** [comment]
I made use of a root_validator with pre=True on the response model itself, to prepare the data.

**Comment by wackazong** [comment]
I would like to do some global exception handling of the route results in a centralized place, but before the response model validation because my service methods return failure objects instead of raising exceptions. Therefore the result of the routes is a Union type [Failure, Model] which I would then like to process in the same way for each route.

**Comment by juntatalor** [comment, marked as answer]
The code which processes handler and response_model validation is here: https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L209

I'm afraid that only one way to achieve desired behavior is to patch the `serialize_response` function (the actual model is passed in `response_field` param) which is not a best practice.

**Comment by wackazong** [comment]
Ok, thanks. I think I will solve it differently then.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @wackazong  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
